[Estuary] Add setting to show profilename

DaVukovic opened the pull request:
<!--- Provide a general summary of your change in the Title above -->
This PR adds a settings to show the profilename next to the clock.

## Description
<!--- Describe your change in detail -->
The profilename is only shown if the profile count is >1 and if the setting is enabled. I also added `max=600` to avoid movement of other items become out of bounds if the profilename is too long.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
If users use additional proifiles, there's currently no way to see which profile is the actual one. Let's say users seperate kids movies from movies 18+ and are using different profiles and different sources for each profile, you will only see at which profile you are if you either look at the shutdownmenu (l'og off' section) or try to compare the movies in the library. This addition changes that 'problem' and users are able to see which profile is in use while browsing the GUI

<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
I tested if the conditions will match. 

I created and deleted profiles to see if the profilename is either shown or not:

if `System.Profilecount > 1` and `Skin.HasSetting(show_profilename)` is enabled, then the profilename is shown

if `System.Profilecount > 1` and `Skin.HasSetting(show_profilename)` is disabled, then the profilename is not shown.

if `System.Profilecount = 1` and `Skin.HasSetting(show_profilename)` is enabled, then the profilename is not shown.

if the profilename is too long, it will be cut and the last characters will be presented with `...`

<!--- Include details of your testing environment, and the tests you ran to -->
I tried compiling on Ubuntu 16.04 and 18.04. I also ziped that skin and installed it as an additional skin (changed IDs for testing purposes) on Kodi installed from the nightly-ppa. 

<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):
![profilename](https://user-images.githubusercontent.com/7235787/41188351-2f621114-6bbc-11e8-84a9-4fe19674c77d.png)

![profilename2](https://user-images.githubusercontent.com/7235787/41188354-3ed09328-6bbc-11e8-8f2c-e7b62474b97c.png)

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [x] I have updated the documentation accordingly
       ^^ I will edit the wiki myself if that change will be merged.
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

DaVukovic committed:
Add functionality to show profile name

da-anda commented:
while I like the idea, the current implementation IMO doesn't look that nice visually. IMO the profile name should be much smaller, so not the same size as the clock but rather the size of the settings label on the left. Even better might be to just show the avatar without profile name, if an avatar is available. Just an idea if you feel like giving it a try.

ksooo commented:
What @da-anda said.

DaVukovic commented:
Sure, I can give it a try. 

Already tried to replace the name with the avatar, but couldn't get it managed ;). Let's see what I can do.

ksooo commented on addons/skin.estuary/language/resource.language.en_gb/strings.po:
```diff
@@ -743,3 +743,8 @@ msgstr ""
+
+#: /xml/SkinSettings.xml
+msgctxt "#31164"
+msgid "Show Profile Name"
```
No upper case, please. => "Show profile name"

ksooo commented on addons/skin.estuary/language/resource.language.en_gb/strings.po:
```diff
@@ -743,3 +743,8 @@ msgstr ""
 msgid "Show Fanart background"
 msgstr ""
+
+#: /xml/SkinSettings.xml
```
Please also add a `#.` line explaining the purpose of this string. In the past we often "forgot" that, but translators will greatly benefit from this information.

DaVukovic commented:
@ksooo 

done. Fine so far?

Still fiddling around with a little 'positioning problem' for the string itself. But I guess I'll ask ronie at slack. Changing the font was easy, but now it's a little step to high. 
But we'll see ;)

ksooo commented on addons/skin.estuary/language/resource.language.en_gb/strings.po:
```diff
@@ -743,3 +743,9 @@ msgstr ""
 msgid "Show Fanart background"
 msgstr ""
+
+#: This string is used at the skin settings for a switch to either show the profile name next to the clock or not
```
If somebody rearranges controls, you might have to change the comment to fit the new location of the label  - and I bet this will be forgotten. I would avoid those "next to something" comments. Describe the item itself, not how it is currently related to something else.

da-anda replied:
since this is a contextual note it has to start with `#.` and not `#:`.
As for the string, how about
`#. Label of a setting that, if enabled, always shows the name of the currently logged-in profile in the UI`

anoma commented:
For what it is worth we have been using the early version of this on four separate devices with 6 profiles for about 2 months now. Even the children can work it and it has made a fundamental positive difference to how we use Kodi "home wide".

DaVukovic commented:
I changed the font to `font45`which is the same as for the settings-breadcrumps on the left

![skin4](https://user-images.githubusercontent.com/7235787/41219283-1f2aa808-6d5e-11e8-92f4-4b1c09671d77.png)
 

The only thing I need to get managed is, that I need to put the string a few steps down. At this position it looks a bit meh

DaVukovic commented:
With the latest changes I used what @da-anda suggests for the dokumentation for the string (thx for that suggestion).
I also fixed a quoting issue I had at `<width max="600">` where the quotes have been missing. 
I changed the font as mentioned above and I also changed the position of the text in the way that it's located on the same position as the clock.

Skin UI now looks like:

![skin5](https://user-images.githubusercontent.com/7235787/41224838-b20c40aa-6d6d-11e8-801b-061d65f77a99.png)
![skin6](https://user-images.githubusercontent.com/7235787/41224839-b229ccce-6d6d-11e8-9254-34962cd80c36.png)

An avatar seems not to be possible currently as there's no info label for it (yet) and it seems to be only available at the profiles settings page. 

That's it for the moment. Being open for any further suggestion ;)

DaVukovic commented:
Lots of changes happened. 

I decided to add both functionalities and let the user choose which one he likes best

Screenshots might say more than 1000 words:

The new setting section with the avatar being shown:
![skin1](https://user-images.githubusercontent.com/7235787/41308733-10d3035c-6e7d-11e8-919d-0eee90e81b52.png)

The matchin dialog:
![skin2](https://user-images.githubusercontent.com/7235787/41308748-1d8008de-6e7d-11e8-8e34-370cbe955b80.png)

Selecting "Profile name":
![skin3](https://user-images.githubusercontent.com/7235787/41308803-3b8f09b0-6e7d-11e8-9683-7a0ff26a893d.png)

Default Icon of "Master user":
![skin4](https://user-images.githubusercontent.com/7235787/41308827-4deddd2a-6e7d-11e8-8449-0d9ec698d532.png)

Hide avatar and name while something is playing:
![skin5](https://user-images.githubusercontent.com/7235787/41308862-60147e5a-6e7d-11e8-891d-aa2ba7a14b4e.png)


Let me know what you will think about it ;)

chewitt commented on addons/skin.estuary/language/resource.language.en_gb/strings.po:
```diff
@@ -743,3 +743,21 @@ msgstr ""
+#. Label of a setting that, if enabled, shows the name of the currently logged-in profile in the UI
+#: /xml/SkinSettings.xml
+msgctxt "#31164"
+msgid "Choose kind of profile identification"
```
Show profile identifier

DaVukovic replied:
True ;)

Will do, thx

ksooo replied:
'Show' implies a on/off functionality. This label is for a choice: profile name, icon, ... So, "show" seems not correct to me.

jjd-uk replied:
Think I agree with ksooo, so perhaps it could be "Choose profile identifier"

DaVukovic replied:
Will use that ^^

chewitt commented on addons/skin.estuary/language/resource.language.en_gb/strings.po:
```diff
@@ -743,3 +743,21 @@ msgstr ""
+#. Label for the kind of profile identification
+#: /xml/SkinSettings.xml
+msgctxt "#31165"
+msgid "Profile name"
```
Name

DaVukovic replied:
If you look at other labels (even those with 2 words) every other label has the 2nd word written in lower case. 

So, tbh not too sure if we should change that 'style'.

ksooo replied:
Keep it as it is now.

jjd-uk replied:
Yes for the settings labels we only ever have the first letter of the first word as a capital, everything other than the first word always begins in lowercase.

chewitt commented on addons/skin.estuary/language/resource.language.en_gb/strings.po:
```diff
@@ -743,3 +743,21 @@ msgstr ""
+#. Label for the kind of profile identification
+#: /xml/SkinSettings.xml
+msgctxt "#31166"
+msgid "Profile avatar"
```
Avatar

DaVukovic replied:
Same as for the label above

ksooo replied:
Okay as it is.

da-anda commented:
nice work. Thanks. My only remark would be that name and avatar stick too close so the time. Could you apply the offset from the playback info to these labels as well?

DaVukovic commented:
Did the requeted changes:

- offset for "avatar" and "name" is the same as for the "now playing" info
added a new 'group' for that as positioning labels was easier that way

- changed the docu for label 31164

![skin6](https://user-images.githubusercontent.com/7235787/41366514-9eca3bca-6f3c-11e8-96e6-26eaa7f210ea.png)
![skin7](https://user-images.githubusercontent.com/7235787/41366515-9eff0242-6f3c-11e8-8421-c46a900e15fd.png)
![skin8](https://user-images.githubusercontent.com/7235787/41366516-9f23ea58-6f3c-11e8-9912-11f4101d28da.png)

kekePower commented:
Nice work @DaVukovic. I dig it.

DaVukovic commented:
Simplified the code a bit. Thanks @da-anda for the suggestion :+1: 

I removed a unnecessary group and changed some bits of the visible-tags. Generally only better readable. The function itself hasn't changed. So no screenshots necessary.

ronie commented on addons/skin.estuary/xml/SkinSettings.xml:
```diff
@@ -67,6 +67,13 @@
+                                <control type="button" id="707">
+                                        <label>$LOCALIZE[31164]</label>
+                                        <include>DefaultSettingButton</include>
+                                        <onclick>Skin.SelectBool(31164, 31165|show_profilename, 31166|show_profileavatar, 16018|)</onclick>
```
please add show_none at the end. SelectBool() needs a setting id for each option.
i'm kinda surprised it even works as is :-)

DaVukovic replied:
done

DaVukovic commented:
done. 

But then we have to fix this line as well: 
https://github.com/xbmc/xbmc/blob/master/addons/skin.estuary/xml/SkinSettings.xml#L67
;)

Rechi commented:
ready for merging?

DaVukovic commented:
From my side, yes. Unless anyone might still have any objections. 

If so, please let me know. I'll happily change necessary things.

hudokkow commented:
No objections. Looks great!

da-anda commented:
also fine from my end. If skinners are happy with the code, hit the button

hudokkow commented on addons/skin.estuary/xml/Includes.xml:
```diff
@@ -893,6 +893,31 @@
+                                                        <animation effect="fade" start="100" end="0" time="300" condition="Window.Next(screencalibration)">WindowClose</animation>
+                                                        <label>$INFO[System.ProfileName]</label>
+                                                </control>
+					</control>
```
Wrong indent

hudokkow commented on addons/skin.estuary/xml/SkinSettings.xml:
```diff
@@ -67,6 +67,13 @@
+                                        <label2>$VAR[ProfileIdentificationLabel2Var]</label2>
+                                </control>
+
 			</control>
```
Wrong indent

hudokkow commented on addons/skin.estuary/xml/Variables.xml:
```diff
@@ -141,6 +141,11 @@
+                <value condition="Skin.HasSetting(show_profileavatar)">$LOCALIZE[31166]</value>
+                <value>$LOCALIZE[16018]</value>
+        </variable>
 	<variable name="AddonsListIconVar">
```
Wrong indent

DaVukovic commented:
Should be fixed now. Thanks for pointing that one out

hudokkow commented:
@ronie good to go?

ronie commented:
it is
